src/context.rs

```diff
@@ -566,6 +569,26 @@ impl PySessionContext {
         Ok(())
     }
 
+    /// Construct datafusion dataframe from Arrow Table
+    pub fn register_table_provider(
+        &mut self,
+        name: &str,
+        provider: Bound<'_, PyAny>,
```

Comment:
Just to understand, taking delta-rs as an example.

The provider would be an instance of a deltatable object?

Reply:
Here is what you would need to implement in `delta-rs` as a python method on `RawDeltaTable`

```
    fn __datafusion_table_provider__<'py>(&self, py: Python<'py>) -> PyResult<Bound<'py, PyCapsule>> {
        let name = CString::new("datafusion_table_provider").unwrap();
        let provider = FFI_TableProvider::new(Arc::new(self._table.clone()));
        PyCapsule::new_bound(py, provider, Some(name.clone()))
    }
```

Does that answer the question?

Reply:
Definitely!